### ISSUE TITLE
Possible bugs

### ISSUE THREAD

**Issue by Mike-Reddick** [jingyu9575/multithreaded-download-manager#135]
1-When add links without multi-threded support MDM automatically recognize and start using 1 connection, but sometimes can't recognize, show the red X and download can't start , then need edit and set manually to 1 connection and then Start, is possible recognize this links also? This happen for instance with some File Host Services.

2-Download sometimes stop, but the progress bar continue blue, need pause-start to resume. Note: Using "legacy storage API"

3- sometimes when a download complete, the save process fail (blue folder), and show a red X, to write the file need click start one time 

4- Now an question is possible implement something to wait internet connection back, for instance when have some downloads running, and you disconnect the wifi, keep download active, and when you reconnect, download continue automatically. Is it possible with the current API?

**Comment by Mike-Reddick** [comment]
@jingyu9575 please see this download links:

https://downloads.novirusthanks.org/files/portables/filegovernor_portable.zip

MDM fail to download, but if edited, setting to 1 connection, download complete. This kind of error isn't rare in MDM, I think a good fix to it can be so, ever when a download fail try again using just 1 connection automatically.

Another link with this issue:

https://www.mexashare.com/5t1gykjkfsse/TrueAnal.Lola.Fae.720p.mp4.html

sorry for the porn link, but is just an example.

Note: Each link have different errors, and both can be fixed with 1 connection

**Comment by jingyu9575** [comment]
> MDM fail to download, but if edited, setting to 1 connection, download complete.

Yes the link's multiple connection support is broken. Firefox 56's DownThemAll will automatically switch to 1 connection. I think the best way is to implement a similar detection and auto-switch.
